Helper/PaymentResponseHandler.php

```diff
@@ -279,6 +290,61 @@ public function handlePaymentsDetailsResponse(
                 break;
             case self::REFUSED:
             case self::CANCELLED:
+                $getGiftcardDetails = $this->hasActiveGiftCardPayments(
+                    $paymentsDetailsResponse['merchantReference']
+                );
+                if (!empty($getGiftcardDetails)) {
+                    //Cancel the Authorised Payments
+                    $storeId = $order->getStoreId();
+                    $client = $this->dataHelper->initializeAdyenClient($storeId);
+                    $service = $this->dataHelper->initializeOrdersApi($client);
+                    foreach ($getGiftcardDetails as $giftcardData) {
+                        try {
+                            // Decode JSON response and validate it
+                            $response = json_decode($giftcardData['response'], true);
+                            if (json_last_error() !== JSON_ERROR_NONE || !isset($response['order'])) {
+                                throw new InvalidArgumentException('Invalid giftcard response data');
+                            }
+
+                            // Extract order data and PSPRef
+                            $orderData = $response['order']['orderData'] ?? null;
+                            $pspReference = $response['order']['pspReference'] ?? null;
+
+                            if (!$orderData || !$pspReference) {
+                                throw new InvalidArgumentException('Missing orderData or pspReference in the response');
+                            }
+
+                            // Prepare cancel request
+                            $merchantAccount = $this->configHelper->getAdyenAbstractConfigData("merchant_account", $storeId);
+                            $cancelRequest = [
+                                'order' => [
+                                    'pspReference' => $pspReference,
+                                    'orderData' => $orderData,
+                                ],
+                                'merchantAccount' => $merchantAccount,
+                            ];
+
+                            // Call the cancel service
+                            $cancelResponse = $service->cancelOrder(new CancelOrderRequest($cancelRequest));
+                            $response = $cancelResponse->toArray();
+
+                            if (is_null($response['resultCode'])) {
+                                // In case the result is unknown we log the request and don't update the history
+                                $this->adyenLogger->error(
+                                    "Unexpected result query parameter for cancel order request. Response: " . json_encode($response)
+                                );
+
+                                return false;
```

Comment:
Returning `false` here will terminate the execution of the rest of the method where we cancel the Magento order. In addition, checking the validity of a single field of an API response is not scalable.

Alternatively, you can catch the exception thrown from the PHP API Library and log the error message only. For responseCode 200, there is only one available value in the [API Explorer](https://docs.adyen.com/api-explorer/Checkout/71/post/orders/cancel#responses-200) which is `resultCode: Received`. So, if there is no exception thrown from the library, we can assume everything is good.